.github/workflows/coverage-check.yaml

```diff
@@ -0,0 +1,136 @@
+name: Coverage Report
+on:
+  workflow_dispatch: # Manual trigger
+  push:
+    branches:
+      - master
+  pull_request:
```

Comment:
What are the benefits of seeing the test coverage in each PR? Only a few engineers are writing tests.

Reply:
What's your suggestion?

Reply:
I revised this logic. The check won't start automatically right now. If you want to run this check on PR, you should add the `Coverage` label to it. Alternatively, it can be run manually.

Reply:
> I revised this logic. The check won't start automatically right now. If you want to run this check on PR, you should add the `Coverage` label to it. Alternatively, it can be run manually.

Nobody will do that. There is absolutely no issue with running coverage on every PR. At least until Microsoft/GitHub start charging us for "free" CI minutes. You can exclude non-C++ stuff from paths in GitHub Actions.

Reply:
There is an issue, sometimes we need to wait for free available CI agents. Faster CI feedback is important.
I propose to run this action only if some tests have changed. It _may_ be useful only if the PR description is updated.

Reply:
Let's add path-ignores:

https://github.com/organicmaps/organicmaps/blob/0b5ba1ba7b2eb0bca265a97f8b869a972fd2a1ef/.github/workflows/linux-check.yaml#L7C1-L29C17


Even better – just merge the coverage into linux-check.yaml. It should not add any significant overhead, maybe 5%.

Reply:
There's no reason to add path-ignores when the workflow cannot be triggered automatically by default. With the current implementation, it can only be triggered in a pull request when the `Coverage` label is set. And I think that's the most optimal solution. There is no need to trigger it in all PRs.

I don't like the idea of adding it to linux-check because coverage requires building sources with additional compile flags. This would necessitate a separate job (similar to the no-unity job), resulting in a larger workflow file. Instead, we can split it into two files (current solution): one for all build/test Linux jobs and one for the optional coverage check job.